packages/flutter_test/lib/src/animation_sheet.dart

```diff
@@ -0,0 +1,345 @@
+
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+/// Records the frames of an animating widget, and later displays the frames as a
+/// grid in an animation sheet.
+///
+/// This class does not support Web, because taking screenshots is unsupported,
+/// which is necessary to build an animation sheet. Tests that use this class
+/// must be noted with `skip: isBrowser`.
+/// (https://github.com/flutter/flutter/issues/56001)
+///
+/// Using this class takes the following steps:
+///
+///  * Create an instance of this class.
+///  * Pump frames that render the target widget wrapped in [record]. Every frame
+///    that has `recording` being true will be recorded.
+///  * Optionally, adjust the size of the test viewport to the [sheetSize] (see
+///    the documentation of [sheetSize] for more information).
+///  * Pump a frame that renders [display], which shows all recorded frames in an
+///    animation sheet, and can be matched against the golden test.
+///
+/// {@tool snippet}
+/// The following example shows how to record an animation sheet of an [Inkwell]
+/// being pressed then released.
+///
+/// ```dart
+/// testWidgets('Inkwell animation sheet', (WidgetTester tester) async {
+///   // Create instance
+///   final AnimationSheetBuilder animationSheet = AnimationSheetBuilder(size: const Size(48, 24));
+///
+///   final Widget target = Material(
+///     child: Directionality(
+///       textDirection: TextDirection.ltr,
+///       child: InkWell(
+///         splashColor: Colors.blue,
+///         onTap: () {},
+///       ),
+///     ),
+///   );
+///
+///   // Optional: setup before recording (`recording` is false)
+///   await tester.pumpWidget(animationSheet.record(
+///     recording: false,
+///     child: target,
+///   ));
+///
+///   final TestGesture gesture = await tester.startGesture(tester.getCenter(find.byType(InkWell)));
+///
+///   // Start recording (`recording` is true)
+///   await tester.pumpFrames(animationSheet.record(
+///     target,
+///     recording: true,
+///   ), const Duration(seconds: 1));
+///
+///   await gesture.up();
+///
+///   await tester.pumpFrames(animationSheet.record(
+///     target,
+///     recording: true,
+///   ), const Duration(seconds: 1));
+///
+///   // Optional: adjust view port size
+///   tester.binding.setSurfaceSize(animationSheet.sheetSize());
+///
+///   // Display
+///   final Widget display = await animationSheet.display();
+///   await tester.pumpWidget(display);
+///
+///   // Compare against golden file
+///   expect(
+///     find.byWidget(display),
+///     matchesGoldenFile('inkwell.press.animation.png'),
+///   );
+/// }, skip: isBrowser); // Animation sheet does not support browser https://github.com/flutter/flutter/issues/56001
+/// ```
+/// {@end-tool}
+class AnimationSheetBuilder {
```

Comment:
Maybe above this mention how this can be used for golden file testing, and then a `See also...` section that links the related docs, like GoldenFileComparator etc.

Reply:
I added "See also" to GoldenFileComparator, but I don't know what you mean by mentioning "how this can be used for golden file testing", since the doc mentioned golden testing in the 4th step as well as the snippet. Can you give an example?

Reply:
Oh right! That should do then! Thanks!